File: Yamore/Yamore.Model/Route.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Yamore.Model
{
    public class Route
    {
        public int RouteId { get; set; }

        public int YachtId { get; set; }

        public int StartCityId { get; set; }

        public int EndCityId { get; set; }

        public int? EstimatedDurationHours { get; set; }

        public string? Description { get; set; }
    }
}

File: Yamore/UI/yamore_mobile/lib/models/route.dart
class RouteModel {
  final int routeId;
  final int yachtId;
  final int startCityId;
  final int endCityId;
  final int? estimatedDurationHours;
  final String? description;

  RouteModel({
    required this.routeId,
    required this.yachtId,
    required this.startCityId,
    required this.endCityId,
    this.estimatedDurationHours,
    this.description,
  });

  factory RouteModel.fromJson(Map<String, dynamic> json) {
    return RouteModel(
      routeId: json['routeId'] as int,
      yachtId: json['yachtId'] as int,
      startCityId: json['startCityId'] as int,
      endCityId: json['endCityId'] as int,
      estimatedDurationHours: json['estimatedDurationHours'] as int?,
      description: json['description'] as String?,
    );
  }
}

